### PR TITLE
default.nix: make JS optimization level configurable

### LINKED ISSUE #229: Make closure compiler level configurable
The closure compiler's ADVANCED optimizations causes problems when using external JS libs. It's been an issue for me when trying to use [reflex-dom-ace](https://github.com/reflex-frp/reflex-dom-ace). It worked fine when using `ob run`, but not at all when I built the frontend with `ghcjs`.

### PR DESCRIPTION
**Pull request by vaibhavsagar**
Fixes #229. We also need this for one of our projects because of weird interactions between this setting and the ACE editor.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,6 +1,7 @@
 { system ? builtins.currentSystem
 , profiling ? false
 , iosSdkVersion ? "10.2"
+, jsOptimizationLevel ? "ADVANCED"
```
Comment (ryantrinkle) [inline default.nix]: Shouldn't this be a project setting, not an Obelisk setting?

**Comment by vaibhavsagar** [comment]
This doesn't work yet, even though the checks pass.

**Comment by ryantrinkle** [comment]
Please mark "WIP"

**Comment by luigy** [comment]
Thanks!

**Pull request merged**